### PR TITLE
Tx 13385 docx ampersand fix

### PR DESCRIPTION
**Pull request by kbairak**
Problem and/or solution
-----------------------

The issue is that BeautifulSoup does a "best effort" when converting its input to XML data. So, when a plain '&' is in the string, the information will be lost:

    >>> BeautifulSoup('<w>a & b</w>', 'xml').w.string
    <<< u'a  b'

This commit adds a `.replace('&', '&amp;')` to the constructor of the soup object used for translations to fix this issue

How to test
-----------

Add a '&' to a translation in the editor. Download the file in `devel` and in this branch.

Reviewer checklist
------------------

Code:
* [ ] Change is covered by unit-tests
* [ ] Code is well documented, well styled and is following [best practices](https://tem.transifex.com)
* [ ] Performance issues have been taken under consideration
* [ ] Errors and other edge-cases are handled properly

PR:
* [ ] Problem and/or solution are well-explained
* [ ] Commits have been squashed so that each one has a clear purpose
* [ ] Commits have a proper commit message [according to TEM](https://tem.transifex.com/github-guide.html#working-on-a-feature)

### CODE REVIEW

File: openformats/formats/docx.py [hunk 1]
```diff
@@ -322,16 +329,22 @@ def compile(self, template, stringset, **kwargs):
...
+            # Do escaping: BeautifulSoup doesn't like unescaped '&' or '<' in
+            # its input. We do expect some '<tx>' and `</tx>` so we only escape
+            # '<' if it's not part of '<tx>'
+            translation = translation.replace(u"&", u"&amp;")
+            translation = re.sub(self.LT_PATTERN, u"&lt;", translation)
+
             translation_soup = BeautifulSoup(
-                u'<wrapper>{}</wrapper>'.format(translation), 'xml'
+                u'<wrapper>{}</wrapper>'.format(translation), 'xml',
             ).find_all(text=True)
```
Comment (igavriil) [inline openformats/formats/docx.py:340]: Instead of custom xml escaping, for some characters, I would prefer an approach like this:
```
from xml.sax.saxutils import escape

translation = translation.replace(u"<tx>", u"__OPENING_TAG__").replace(u"</tx>", u"__CLOSING_TAG__")
translation = escape(translation)
translation = translation.replace(u"__OPENING_TAG__", u"<tx>").replace(u"__CLOSING_TAG__", u"</tx>")
```

what do you think?

**Review by igavriil — COMMENTED** [review]
Looks great!
I would prefer a different approach, since I am not a fun of regexes. Let me know what you think

File: openformats/formats/docx.py (hunk 1 above)
Comment (kbairak) [reply openformats/formats/docx.py:340]: I like not having to use regex, but I don't like bringing in another library to help with escaping. I'll look into what can be done with using BeautifulSoup for escaping. If I can't figure out anything, I will use yours.

Comment (kbairak) [reply openformats/formats/docx.py:340]: It doesn't work because `<tx>` tags can have attributes. Even if I replace with `__OPENING_TAG__` only the `<tx` part, then the escape function will escape the closing `>` of the tag.

Reading the docs of `xml.sax.saxutils`, the `escape` function does very simple replacements. So we could keep your idea, replace only the `<tx` part and do the escaping with `.replace`, only replacing the `<` and `&` characters since apparently BeautifulSoup is ok with the `>` ones.

Comment (kbairak) [reply openformats/formats/docx.py:340]: done

**Review by codegaze — APPROVED** [review]
